driver/src/price_finding/naive_solver.rs

```diff
@@ -0,0 +1,349 @@
+use web3::types::U256;
+
+use crate::models::{Order, State, TOKENS};
+
+use super::price_finder_interface::{PriceFinding, Solution};
+use crate::price_finding::error::PriceFindingError;
+
+pub enum OrderPairType {
+    LhsCompletelyFulfilled,
+    RhsCompletelyFulfilled,
+    BothPartiallyFulfilled,
+}
+
+impl Order {
+    fn attracts(&self, other: &Order) -> bool {
+        self.opposite_tokens(other) && self.have_price_overlap(other)
+    }
+
+    fn sufficient_seller_funds(&self, state: &State) -> bool {
+        let balance_index = self.sell_token as usize + (self.account_id - 1) as usize * TOKENS as usize;
```

Comment:
```suggestion
        let balance_index = (self.sell_token as usize) - 1 + (self.account_id - 1) as usize * TOKENS as usize;
```

Reply:
I think we should somehow make a balance index function as a utility somewhere.

Reply:
Yes please. Let’s do a PR where balances is no longer public on the struct so that no one can directly access it, but instead has to go through helper functions.